src/typing_manager.ts

```diff
@@ -72,14 +72,14 @@ export class TypingManager implements Disposable {
                 );
                 this.changeManager.getDocumentChangeCompletionLock(editor.document)?.then(() => {
                     this.isEnteringInsertMode = false;
-                    if (this.typeHandlerDisposable) {
+                    if (this.typeHandlerDisposable && this.modeManager.isInsertMode) {
                         this.logger.debug(`${LOG_PREFIX}: Waiting done, disposing type handler`);
                         this.typeHandlerDisposable.dispose();
                         this.typeHandlerDisposable = undefined;
-                        if (this.pendingKeysAfterEnter) {
-                            commands.executeCommand("default:type", { text: this.pendingKeysAfterEnter });
-                            this.pendingKeysAfterEnter = "";
-                        }
+                    }
+                    if (this.pendingKeysAfterEnter) {
+                        commands.executeCommand("default:type", { text: this.pendingKeysAfterEnter });
```

Comment:
I forgot - need one more thing here, executing `default:type` may insert text in normal mode which is wrong,

Change to:
`commands.executeCommand(this.modeManager.isInsertMode ? "default:type" : "type", ....)`

Reply:
Thanks!